hazelcast/include/hazelcast/client/reliable_topic.h

```diff
@@ -415,6 +420,7 @@ class HAZELCAST_API reliable_topic : public proxy::ProxyImpl
         serialization::pimpl::SerializationService& serialization_service_;
         int batch_size_;
         util::SynchronizedMap<int, util::concurrent::Cancellable>& runners_map_;
+        std::shared_ptr<reliable_topic> topic_;
```

Comment:
Could you make it `weak_ptr` since it is not required to be alive to continue work.

Reply:
Shared_ptr is changed with weak_ptr and runners_map_ is removed from MessageRunner. It is access via topic_ pointer.